src/main/java/com/minecolonies/coremod/colony/Structures.java

```diff
@@ -60,200 +109,845 @@ public static void init()
 
     /**
      * Loads all styles saved in ["/assets/minecolonies/schematics/"].
-     * Puts these in {@link #hutStyleMap}, with key being the name of the hutDec (E.G. Lumberjack).
-     * and the value is a list of styles. Puts decorations in {@link #decorationStyleMap}.
+     * Puts these in {@link #md5Map}, with key being the fullname of the structure (schematics/stone/Builder1).
      */
     private static void loadStyleMaps()
     {
-        try
+        if (!Configurations.ignoreSchematicsFromJar)
         {
-            @NotNull final URI uri = ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI();
-            final Path basePath;
-
-            if ("jar".equals(uri.getScheme()))
+            try
             {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap()))
+                @NotNull final URI uri = ColonyManager.class.getResource(SCHEMATICS_ASSET_PATH).toURI();
+                FileSystem fileSystem;
+                try
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1141)